src/main/java/org/myrobotlab/service/Runtime.java

```diff
@@ -133,7 +136,7 @@ public class Runtime extends Service<RuntimeConfig> implements MessageListener,
    * a registry of all services regardless of which environment they came from -
    * each must have a unique name
    */
-  static private final Map<String, ServiceInterface> registry = new TreeMap<>();
+  static private Map<String, ServiceInterface> registry = new TreeMap<>();

```

Comment:
Why can't this be final anymore?

Reply:
ConcurrentHashMap isn't sorted TreeMap is, to make it threadsafe removal actions need to re-new it